bigquery/google/cloud/bigquery/enums.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+import warnings
+
+import enum
+import six
+
+from google.cloud.bigquery_v2.gapic import enums as gapic_enums
+
+
+def _make_sql_scalars_enum():
+    """Create an enum based on a gapic enum containing only SQL scalar types."""
+    sql_scalar_types = frozenset(
+        (
+            "INT64",
+            "BOOL",
+            "FLOAT64",
+            "STRING",
+            "BYTES",
+            "TIMESTAMP",
+            "DATE",
+            "TIME",
+            "DATETIME",
+            "NUMERIC",
+        )
+    )
+    excluded_members = frozenset(
+        ("TYPE_KIND_UNSPECIFIED", "GEOGRAPHY", "ARRAY", "STRUCT")
+    )
+
+    # Sanity check - we do not want the new enum to go out of sync with the original
+    # enum from gapic.
+    # ASSUMPTION: No existing types are ever renamed or deleted, we only try to
+    # detect cases when new types are introduced.
+    gapic_names = set(m.name for m in gapic_enums.StandardSqlDataType.TypeKind)
```

Comment:
Nit: We use descriptive names even in set comprehensions, since `m` leaks into the local function scope. How about `type_`, instead?

Reply:
Leaking only occurs in _list_ comprehensions, and only in Python 2, but I agree, will use a nicer name. In a month even I probably won't remember that "m" stands for "member". :smile: